src/classes/BDI_Donations.cls

```diff
@@ -236,22 +244,55 @@ public class BDI_Donations {
 
         // add any new payments to our map
         trackPayments(dataImportDonations);
-        
+
         // when we mark Payments paid, ensure they will close our Opp.
         npe01__Contacts_And_Orgs_Settings__c cs = UTIL_CustomSettingsFacade.getContactsSettings();
         if (cs.Payments_Auto_Close_Stage_Name__c == null) {
             cs.Payments_Auto_Close_Stage_Name__c = defaultClosedWonStage;
         }
-        
+
         // update payments with any data from the DI record
         updatePayments(dataImportDonations);
     }
 
+    /*******************************************************************************************************
+    * @description Determine if the data import batch number should be set on the opportunity or payment
+    * @param targetObject The opportunity or payment SObject record
+     * @param dataImport The data import record being processed
+    * @return Boolean Whether or not the DI batch number should be set on opportunities or payments
```

Comment:
One thing I like for documenting methods that return booleans is to write out the criteria that causes a certain value, usually TRUE, to be returned.

`* @return Boolean TRUE when the Data Import Batch has a batch number, is not a dry run, and the current  SObject is an Opportunity or Opportunity Payment`

Reply:
good idea!